scripts/validate_string_concatenation.py

```diff
@@ -1,129 +1,377 @@
 #!/usr/bin/env python3
 """
-GH #30454
+Unwanted patterns test cases.
 
-Check where there is a string that needs to be concatenated.
+The reason this file exist despite the fact we already have
+`ci/code_checks.sh`,
+(see https://github.com/pandas-dev/pandas/blob/master/ci/code_checks.sh)
 
-This is necessary after black formating,
-where for example black transforms this:
-
->>> foo = (
-...         "bar "
-...         "baz"
-...     )
-
-into this:
-
->>> foo = ("bar " "baz")
-
-Black is not considering this as an
-issue (see issue https://github.com/psf/black/issues/1051),
-so we are checking it here.
+is that some of the test cases are more complex/imposible to validate via regex.
+So this file is somewhat an extensions to `ci/code_checks.sh`
 """
 
 import argparse
 import os
 import sys
 import token
 import tokenize
-from typing import Generator, List, Tuple
+from typing import IO, Callable, Generator, List, Tuple
+
+FILE_EXTENSIONS_TO_CHECK: Tuple[str, ...] = (".py", ".pyx", ".pyx.ini", ".pxd")
```

Comment:
```suggestion
FILE_EXTENSIONS_TO_CHECK: Tuple[str, ...] = (".py", ".pyx", ".pxi.in", ".pxd")
```

Reply:
@WillAyd This apparently breaks the script.